### PR TITLE
cleanup: harmonise _is_valid methods (followup #2695)

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

Follow-up PR to #2695 :  update all `_is_valid()` and `__new__()` implementations for `Dataset` children classes.

note: this will conflict with https://github.com/yt-project/yt/pull/2807, and should be rebased after it goes in, leaving this as a draft in the mean time

## todo
- [x] replace `args[0]` with named positional argument `filename` in
  * [x] `<DatasetChild>._is_valid`
  * [x] `<DatasetChild>.__init__` (nothing to be done :tada:)
  * [x] `<DatasetChild>.__new__`
  * [x] update `<DatasetChild>._is_valid`'s first positional argument to `cls` (it's a class method but currently some implementations incorrectly name the first arg `self`) (not really thematically linked to this PR but while I'm at it, might as well fix this)
- [x] update `YTOutputNotIdentified`

### CODE REVIEW

**Comment by neutrinoceros** [comment]
switching to draft to clean up the PR backlog

**Comment by neutrinoceros** [comment]
I don't want to be rebasing this for ever. Switching to draft until #2695 is dealt with.

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Comment by neutrinoceros** [comment]
switching to draft to avoid collisions with https://github.com/yt-project/yt/pull/2746

**Comment by neutrinoceros** [comment]
This has become virtually impossible to rebase. Closing.

**Comment by neutrinoceros** [comment]
Oops, wrong branch, this one is actually ok !

**Review by matthewturk — APPROVED** [review]
I have reviewed this and it looks good to me.  I like the harmonization process.

**Review by matthewturk — APPROVED** [review]
This looks good; any reason we can't merge?

File: yt/frontends/exodus_ii/data_structures.py [hunk 1]
```diff
@@ -394,12 +394,12 @@ def _load_domain_edge(self):
...
     @classmethod
-    def _is_valid(self, *args, **kwargs):
-        warn_netcdf(args[0])
+    def _is_valid(cls, filename, *args, **kwargs):
+        warn_netcdf(filename)
         try:
             from netCDF4 import Dataset
 
-            filename = args[0]
+            filename = filename
```
Comment (cphyc) [inline yt/frontends/exodus_ii/data_structures.py:402]:
```suggestion
```

File: yt/frontends/enzo_p/data_structures.py [hunk 2]
```diff
@@ -465,8 +465,8 @@ def __repr__(self):
         return self.basename[: -len(self._suffix)]
 
     @classmethod
-    def _is_valid(cls, *args, **kwargs):
-        fn = args[0]
+    def _is_valid(cls, filename, *args, **kwargs):
+        fn = filename
```
Comment (cphyc) [inline yt/frontends/enzo_p/data_structures.py:469]:
```suggestion
```
Note: you'll have to update line 474 below as well.

File: yt/frontends/rockstar/data_structures.py [hunk 3]
```diff
@@ -112,10 +112,10 @@ def _set_code_unit_attributes(self):
         setdefaultattr(self, "time_unit", self.length_unit / self.velocity_unit)
 
     @classmethod
-    def _is_valid(self, *args, **kwargs):
-        if not args[0].endswith(".bin"):
+    def _is_valid(cls, filename, *args, **kwargs):
+        if not filename.endswith(".bin"):
             return False
-        with open(args[0], "rb") as f:
+        with open(filename, "rb") as f:
```
Comment (cphyc) [inline yt/frontends/rockstar/data_structures.py:118]:
```suggestion
        with open(filename, mode="rb") as f:
```

**Review by cphyc — APPROVED** [review]
Looks good to me! I'm ok to merge once the comments are included

**Pull request merged**